src/bif/map.rs

```diff
@@ -44,33 +55,85 @@ pub fn bif_maps_is_key_2(_vm: &vm::Machine, process: &RcProcess, args: &[Value])
         let exist = hamt_map.find(target).is_some();
         return Ok(Value::boolean(exist))
     }
-    let heap = &process.context_mut().heap;
-    let tuple = tup2!(&heap, atom!(BADMAP), map.clone());
-    Err(Exception::with_value(Reason::EXC_BADMAP, tuple))
+    Err(Exception::with_value(Reason::EXC_BADMAP, map.clone()))
 }
 
-pub fn bif_maps_keys_1(_vm: &vm::Machine, _process: &RcProcess, args: &[Value]) -> BifResult {
-    unimplemented!();
+pub fn bif_maps_keys_1(_vm: &vm::Machine, process: &RcProcess, args: &[Value]) -> BifResult {
+    let map = &args[0];
+    if let Value::Map(m) = map {
+        let hamt_map = &m.0;
+        let heap = &process.context_mut().heap;
+        let mut vec = vec![];
+        for (key, _) in hamt_map.iter() {
```

Comment:
I cannot use $iter.rev() with HamtMapIterator.
The trait `std::iter::DoubleEndedIterator` is not implemented in HamtMapIterator yet.
https://github.com/michaelwoerister/hamt-rs/blob/master/src/hamt.rs#L1460
It only impl the Iterator instead of DoubleEndedIterator

Reply:
Hmm, I see. So I think maps don't have a defined key order anyway, so it's fine if we don't do `.rev()`